Admin/Test-AMSI.ps1

```diff
@@ -1,144 +1,793 @@
+                            Write-Warning  "We have not found activity on the server in the last minute."
+                        }
+                        if ( -not $detected ) {
+                            Write-Warning "We have not found a detection."
+                        }
+                    } else {
+                        Write-Host "We could not access HttpRequestFiltering folder on $server" -ForegroundColor Red
+                    }
+                } else {
+                    Write-Host "Check your log files located in %ExchangeInstallPath%\Logging\HttpRequestFiltering\ in all server that provide $server endpoint"
+                }
+            } elseif ( $_.Exception.Message -eq "The remote server returned an error: (500) Internal Server Error.") {
+                $host.ui.RawUI.ForegroundColor = "Red"
+                Write-Host $msgNewLine
+                Write-Host $Message
+                Write-Host $msgNewLine
+                $host.ui.RawUI.ForegroundColor = "Yellow"
+                Write-Host "If you are using Microsoft Defender, RealTime protection could be disabled or then AMSI may be disabled."
+                Write-Host "If you are using a 3rd Party AntiVirus Product that may not be AMSI capable (Please Check with your AntiVirus Provider for Exchange AMSI Support)"
+                $host.ui.RawUI.ForegroundColor = $currentForegroundColor
+            } elseif ( $_.Exception.Message.StartsWith("The remote name could not be resolved:") ) {
+                $host.ui.RawUI.ForegroundColor = "Red"
+                Write-Host $msgNewLine
+                Write-Host $Message
+                Write-Host $msgNewLine
+                $host.ui.RawUI.ForegroundColor = $currentForegroundColor
+            } else {
+                $host.ui.RawUI.ForegroundColor = "Red"
+                Write-Host $msgNewLine
+                Write-Host $Message
+                Write-Host $msgNewLine
+                $host.ui.RawUI.ForegroundColor = "Yellow"
+                Write-Host "If you are using Microsoft Defender, RealTime protection could be disabled or then AMSI may be disabled."
+                Write-Host "If you are using a 3rd Party AntiVirus Product that may not be AMSI capable (Please Check with your AntiVirus Provider for Exchange AMSI Support)"
                 $host.ui.RawUI.ForegroundColor = $currentForegroundColor
-                Write-Output $msgNewLine
             }
         } finally {
             if ($IgnoreSSL) {
-                SetCertificateValidationBehavior -Default
+                Disable-TrustAnyCertificateCallback
+            }
+        }
+        Write-Host "Ended test at $(Get-Date -Format "yyyy-MM-dd HH:mm:ss")"
+    }
+
+    function GetWindowsMayorVersion {
+        param(
+            [Parameter(Mandatory = $true)]
+            [string]$ExchangeServer
+        )
+
+        if ( $ExchangeServer.ToLower() -eq $env:COMPUTERNAME.ToLower() ) {
```

Comment:
The function can be simplified by just using `Invoke-ScriptblockHandler`. No need to use `Get-CimInstance` or `Get-WmiObject`

`Invoke-ScriptBlockHandler -ComputerName $ExchangeServer -ScriptBlock { [System.Environment]::OSVersion.Version.Major }`

![image](https://github.com/microsoft/CSS-Exchange/assets/40993616/83d75f19-9677-40f2-91f5-1631fef28276)